lib/css/exports/global-modifiers.less

```diff
@@ -0,0 +1,7 @@
+.is-disabled,
```

Comment:
Imo, we should inline this one style anywhere it is used. This is something that could bite us in the rear later on, especially if we're including it in `exports` (essentially making it part of our public LESS api)

Reply:
Ah, I didn't really consider the implications of including this in our `/exports/` directory. Yeah, this really isn't the best home for these classes.

The issue is that these classes have this rule applied without any parent/sibling classes applied, so I'm not sure where the best permanent home would be (besides a file like this one). My first instinct is to move this file to `lib/css/atomic` since they're applied _kinda_ like atomic classes, but really they're modifiers 🤷‍♂️ Any thoughts?

Reply:
Hi guys! Quick note: as you know, this State rules comes from SMACSS approach (http://smacss.com/book/type-state) and they are mostly tied to changing behavior using JS. I think they are good for naming component states (you can see very quickly what is going on in CSS and what variant it is) and IMHO they don't need to be in a separate file because they have no general meaning.

Reply:
If you plan to use for e.g. "is-hidden" or "is-invisible" that can be used everywhere, I think some /atomic class folder should work here.

Reply:
@giamir @b-kelly any ideas wrt [my comment](https://github.com/StackExchange/Stacks/pull/1184#discussion_r1038536330)?

Reply:
I haven't reviewed in details this PR but at a first glance I also find odd to see these classes globally available across the whole document.  As a rule of thumbs we should always think twice before exposing more API surface to consumers because we are implicitly making the system harder for us to change/evolve.
I would prefer if we could keep those classes scoped to the component that needs them (similar to what Ben suggested). This will create a bit of duplication but I would not mind to prioritize encapsulation over performance in this context.
If we have instances where these classes are just used in random contexts then we could put the rules in a place where it's clear we will deprecate them in the next major release (maybe a `pending-deprecation.less` file?) How much work do you foresee in Core for replacing these classes usage?
By the way the deprecation strategy ADR and the css dev bundle experiment is still in my todo list. I should probably create an issue for that but won't be able to tackle it before next year.
I hope this rambling is somewhat useful. 🙂

Reply:
>  If we have instances where these classes are just used in random contexts then we could put the rules in a place where it's clear we will deprecate them in the next major release (maybe a pending-deprecation.less file?) How much work do you foresee in Core for replacing these classes usage?

@dancormier @b-kelly Do you think is possible to deprecate those classes in our next major release? How painful would that be?

Reply:
> How much work do you foresee in Core for replacing these classes usage?

I think it'd be a pretty heavy lift to replace those classes in Core. There's lots of usage of `.is-disabled` and `.has-error` that would need to be replaced. Often, these classes are set with JS conditionally, so time would be needed to understand context and prevent regressions.

> How painful would that be?

I think it would be pretty painful and honestly I feel like this is something we should punt on. In the context of this PR, nothing would materially change for consumers wrt these `.is-` and `.has-` classes. I'm good with being overruled if I'm being too pessimistic, but felt I should voice my opinion.

Reply:
Outside of the deprecation consideration, where could be the best place for these styles @giamir @b-kelly? `lib/css/atomic`?

Reply:
Looking at some of the other code, we may need to keep around an e.g. `input-utils.less` file for mixins. These could likely go in there as well.